http-client/Network/HTTP/Client/Core.hs

```diff
@@ -122,20 +126,31 @@ httpRaw' req0 m = do
             Nothing -> return (req, res)
   where
     getConnectionWrapper mtimeout f =
-        case mtimeout of
+        bracketOnError getConnWithTimeout releaseConnMaybe checkConnDuration
+      where
+        -- If applicable, wait for timeout period to acquire the connection from the pool.
+        getConnWithTimeout = case mtimeout of
             Nothing -> fmap ((,) Nothing) f
             Just timeout' -> do
                 before <- getCurrentTime
                 mres <- timeout timeout' f
```

Comment:
This is a problem. `timeout` is now running in a masked state, which we shouldn't be doing.

Reply:
Should `timeout` function be wrapped in `interruptible` explicitly?

Reply:
@snoyberg Why is it a problem? `getConn` is interruptible, so `timeout` will be able to interrupt it. And in any case, `getConn` basically just calls `takeKeyedPool`, with is `mask_`'ed anyway.

Reply:
I don't want to rely on the current behavior of `bracketOnError` not masking in the acquire clause, that may change in the future. It's fragile to write code that way IMO. It seems like the primary thing we're trying to achieve is ensuring that `releaseConnMaybe` is called in the case of `remainingTime <= 0`. Why not add that in directly before the `throwHttp` call, or add an `onException`?

Reply:
Restored previous version and handled directly. Fixed.